### ISSUE TITLE
Mainline doesn't show Linux Kernel 5.7.2, 5.7.3 and 5.7.4 and 5.8-rc1

### ISSUE THREAD

**Issue by MateuszKrawczuk** [bkw777/mainline#43]
I tried cleaning cache but even after removing everything and reinstalling package doesn't show newer kernels.

**Comment by Deleetdk** [comment]
It's not just the 5.7.X that are missing. It's missing newer versions of the older releases too. Here's a comparison with the results in Ukuu paid.

![Screenshot from 2020-06-22 15-20-47](https://user-images.githubusercontent.com/6759913/85292324-04ffeb00-b49c-11ea-80b8-e47f19582ac3.png)
![Screenshot from 2020-06-22 15-20-35](https://user-images.githubusercontent.com/6759913/85292325-05988180-b49c-11ea-840c-92323f58162b.png)
![Screenshot from 2020-06-22 15-20-25](https://user-images.githubusercontent.com/6759913/85292326-06311800-b49c-11ea-8a16-95522afbd6c5.png)

**Comment by rigoleto** [comment]
I assumed these versions were missing because of failing builds: https://kernel.ubuntu.com/~kernel-ppa/mainline/v5.7.5/

`Test amd64/self-tests failed (rc=2, on=amd64, time=0:01:02, log=amd64/self-tests/log)`

**Comment by bkw777** [comment]
There is definitely something wrong with the way the version strings are parsed and compared against each other in the latest version. I am working on a re-write of all the stuff related to that, but until then I think one version back is still ok. Sorry.

**Comment by TID41** [comment]
The URL of the actual .deb files has changed.  They've added 'amd64' after the version number (for x64 kernels, of course) for everything newer than June 7:

```
https://kernel.ubuntu.com/~kernel-ppa/mainline/v5.7.1/linux-image-unsigned-5.7.1-050701-generic_5.7.1-050701.202006071230_amd64.deb
https://kernel.ubuntu.com/~kernel-ppa/mainline/v5.7.2/**amd64**/linux-image-unsigned-5.7.2-050702-generic_5.7.2-050702.202006101934_amd64.deb
```

**Comment by StatusCode404** [comment]
Hi @bkw777; thanks for the fork and trying to fix the bug!   So has it been fixed?  If so we'll need the ppa updated as well.

Thanks again

**Comment by noah1510** [comment]
It seems like #44 fixes it you can download the source from https://github.com/tovam/mainline and build the package yourself or wait until the pull request is merged.

If you want to manually update the package follow the build instructions of the readme, but instead of running `make` and `make install` run `sudo apt-get install devscripts debhelper`, `debuild -uc -us` and `sudo dpkg -i ../mainline_1.0.12_amd64.deb`. This will build the debian package from source and install it (basically a manual update).

**Comment by StatusCode404** [comment]
Is the PPA automated?  Can we expect the PPA updated soon?

**Comment by noah1510** [comment]
> Is the PPA automated?
I don't really think so although the changelog always says automatic build :man_shrugging: 

>  Can we expect the PPA updated soon?
Probably, however I think there will be a new version soon and after that the ppa should be updated.

**Comment by cappelikan** [comment]
> > Чи PPA автоматизовано?
> 
> Я не думаю, що так, хоча журнал змін завжди говорить про автоматичну збірку 
> 
> > Чи можемо ми очікувати, що скоро буде оновлено PPA?
> 
> Напевно, проте я думаю, що незабаром з'явиться нова версія, після чого ppa має бути оновлений.
> Редагувати: ppa не підтримується bkw777, а не @cappelikan, тому це залежить від того, коли він хоче оновити ppa.

mainline ppa is automatic compile and build deb

**Comment by noah1510** [comment]
Okay good to know, the latest version shows the newer kernels but as I commented on #44 the changes and remove button do not work for those versions or rather cannot be clicked

**Comment by StatusCode404** [comment]
Same for me, I can finally see the newer versions but I can't view the release note "changes".  The button is greyed out.

**Comment by StatusCode404** [comment]
@bkw777 just wondering if you saw our note on the "changes" button being greyed out for the newer kernels?

**Comment by tovam** [comment]
Mainline can't do anything about this problem: the `CHANGES` file aren't published anymore on kernel.ubuntu.com
Before: https://kernel.ubuntu.com/~kernel-ppa/mainline/v5.6.17
After: https://kernel.ubuntu.com/~kernel-ppa/mainline/v5.6.18

**Comment by bkw777** [comment]
We can detect that and disable the button, or display an explaination in place of the file, or fetch the changes from somewhere else.

**Comment by StatusCode404** [comment]
I'll check the mailing list to see why it isn't included anymore.

You can join the ubuntu kernel mailing list here: [https://lists.ubuntu.com/mailman/listinfo/kernel-team](url)
to put pressure on them to sort it out.  Release Notes are always mandatory if you think about it.

**Comment by noah1510** [comment]
What about the remove button? Do you know why it is greyed out?

> Release Notes are always mandatory if you think about it.

I totally agree especially when it comes to core components of an operating system.

> We can detect that and disable the button, or display an explaination in place of the file, or fetch the changes from somewhere else.

+1 for at least showing an explanation.

**Comment by StatusCode404** [comment]
I have asked the question here: https://lists.ubuntu.com/archives/kernel-team/2020-July/111962.html

If someone could reply to that advising that you are also experiencing the same problem then that would hopefully catch one of the team's eyes and add pressure on them to fix it.

**Comment by bkw777** [comment]
Only other kind of changelog I can find, that we could fetch automatically, is the actually full every-single-detail changelog, not a nice summary. Ex:
https://cdn.kernel.org/pub/linux/kernel/v5.x/ChangeLog-5.6.17

**Comment by bkw777** [comment]
actually, looking at the old CHANGES vs the upstream Changelog, looks like the old CHANGES format can be parsed out of the upstream changelog without too much trouble:
Compare a single item from:
https://kernel.ubuntu.com/~kernel-ppa/mainline/v5.6.17/CHANGES
https://cdn.kernel.org/pub/linux/kernel/v5.x/ChangeLog-5.6.17

upstream version:
<pre>
commit 897b7c720803aacf7c94a0054648210d7c737f06
Author: DENG Qingfang <dqfext@gmail.com>
Date:   Wed May 13 23:37:17 2020 +0800

    net: dsa: mt7530: set CPU port to fallback mode
    
    commit 38152ea37d8bdaffa22603e0a5b5b86cfa8714c9 upstream.
    
    Currently, setting a bridge's self PVID to other value and deleting
    the default VID 1 renders untagged ports of that VLAN unable to talk to
    the CPU port:
    
            bridge vlan add dev br0 vid 2 pvid untagged self
            bridge vlan del dev br0 vid 1 self
            bridge vlan add dev sw0p0 vid 2 pvid untagged
            bridge vlan del dev sw0p0 vid 1
            # br0 cannot send untagged frames out of sw0p0 anymore
    
    That is because the CPU port is set to security mode and its PVID is
    still 1, and untagged frames are dropped due to VLAN member violation.
    
    Set the CPU port to fallback mode so untagged frames can pass through.
    
    Fixes: 83163f7dca56 ("net: dsa: mediatek: add VLAN support for MT7530")
    Signed-off-by: DENG Qingfang <dqfext@gmail.com>
    Reviewed-by: Florian Fainelli <f.fainelli@gmail.com>
    Signed-off-by: David S. Miller <davem@davemloft.net>
    Signed-off-by: Greg Kroah-Hartman <gregkh@linuxfoundation.org>
</pre>

old summary version:
<pre>
DENG Qingfang (1):
      net: dsa: mt7530: set CPU port to fallback mode
</pre>

I'll have to look around some more at other versions to see what the (1) is. Maybe the number of items from that same submitter? This particular example might make the job look falsely simpler than it is, just because it happens that in this case every submitter had a single item, where maybe the script that made the old summaries did a little bit of collating and collapsing. That still wouldn't be too bad, and we don't necessarily have to replicate that immeditely. It would probably be fine to just do the simple job of just removing some of the git details, and maybe even keep some of those more verbose descriptions even though they weren't in the old summaries.

In fact, in the real short term, it's probably good enough to just make the button launch the link to the upstream changelog.

**Comment by bkw777** [comment]
Moving the changelog issue to it's own new issue #48

**Comment by bkw777** [comment]
Fixed by https://github.com/bkw777/mainline/commit/9920fabd429a39072801742dfed9e858d6c4479a
